elegant/clean_timepoint_data.py

```diff
@@ -0,0 +1,147 @@
+
+    for dict_entry_type in ['brightfield metering', 'fluorescent metering', 'humidity', 'temperature']:
+        expt_md[dict_entry_type] = {key:val for key,val in expt_md[dict_entry_type].items() if key != timepoint}
+
+    datafile.json_encode_atomic_legible_to_file(expt_md,experiment_root/'experiment_metadata.json')
+
+def remove_excluded_positions(experiment_root,dry_run=False):
+    """Deletes excluded positions from an experiment directory
+
+    Parameters
+        experiment_root - str/pathlib.Path to experiment
+        dry_run - bool flag that toggles taking action (if False, only specifies when offending files are found)
+    """
+
+    experiment_root = pathlib.Path(experiment_root)
+    annotations = load_data.read_annotations(experiment_root)
+    good_annotations = load_data.filter_annotations(annotations, load_data.filter_excluded)
+    excluded_positions = sorted(set(annotations.keys()).difference(set(good_annotations.keys())))
+    for position in excluded_positions:
+        if (experiment_root / position).exists():
+            print(f'Found an excluded position to delete {position}')
+            if not dry_run:
+                shutil.rmtree(str(experiment_root / position))
+
+def remove_dead_timepoints(experiment_root, postmortem_timepoints, dry_run=False):
+    """Deletes excess timepoints in an experiment where worms are dead
+
+    Parameters
+        experiment_root - str/pathlib.Path to experiment
+        postmortem_time - Number of timepoints to keep past the annotated death timepoint;
+            useful for keeping extra timepoints in case one ever wants to validate
+            death for the previously made annotations; postmortem_timepoints should be an int >= 1
+        dry_run - bool flag that toggles taking action (if False, only specifies when offending files are found);
+            this flag is exposed for corresponding functionality through remove_timepoint_for_position
+    """
+
+    if postmortem_timepoints < 1:
+        raise ValueError('postmortem_timepoints should be >= 1')
+    elif type(postmortem_timepoints) is not int:
+        raise ValueError('postmortem_timepoints must be an integer')
+
+    experiment_root = pathlib.Path(experiment_root)
+    annotations = load_data.read_annotations(experiment_root)
+    good_annotations = load_data.filter_annotations(annotations, load_data.filter_excluded)
+
+    for position, position_annotations in good_annotations.items():
+        general_annotations, timepoint_annotations = position_annotations
+        timepoint_stages = [timepoint_info['stage'] for timepoint_info in timepoint_annotations.values()]
+        death_timepoint_index = timepoint_stages.index('dead')
+
+        for timepoint_num, timepoint in enumerate(timepoint_annotations):
+            timepoints_after_death = timepoint_num - death_timepoint_index # positive values for timepoints after death
+            if timepoints_after_death > postmortem_timepoints:
```

Comment:
If for whatever reason `timepoints_after_death` were a float, it would still work fine here... If it were non-numerical, you'd have already gotten an error at `if postmortem_timepoints < 1:`...

Reply:
Deleted it as above; figured that I would prevent the nonsensical but should be abated.